stories/__tests__/SQFormButton.stories.test.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import {composeStories} from '@storybook/testing-react';
+import * as stories from '../SQFormButton.stories';
+
+const {
+  Default: SQFormButton,
+  WithTestField: SQFormButtonWithField
+} = composeStories(stories);
+
+describe('Default Tests', () => {
+  it('should render a button with text', () => {
+    render(<SQFormButton />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    expect(submitButton).toBeInTheDocument();
+    expect(submitButton).toHaveTextContent('Submit');
+  });
+
+  it('should render a reset button given the type', () => {
+    render(<SQFormButton type="reset" />);
+
+    const resetButton = screen.getByRole('button', {name: /form submission/i});
+
+    expect(resetButton).toBeInTheDocument();
+    expect(resetButton).toHaveAttribute('type', 'reset');
+  });
+
+  it('should call a function when clicked', () => {
+    const onClickSpy = jest.fn();
+    render(<SQFormButton onClick={onClickSpy} />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    userEvent.click(submitButton);
+
+    expect(onClickSpy).toHaveBeenCalled();
```

Comment:
Should we specify, one click should be called once?

Reply:
Yeah, ideally I could test that data that was submitted, but I'd have to make some changes to how the current stories are implemented to be able to override onSubmit or to use console.warn or something like that I can mock.
I will update it to be specific to 1 call.

Reply:
I think we can specify what the `spy` is called with right? (The args it should be invoked with)

Reply:
> I think we can specify what the `spy` is called with right? (The args it should be invoked with)

Ah nm, I don't think we can do this because we are in the component story

Reply:
I did attempt it, but the actual args that go to onClick was a lot to try and match. The SQForm.stories ones that we check the args for are with console.warn

Reply:
I think that would be implementation details territory. Here we're testing that the onClick is set up properly - what it's called with is dependent on how/where the button is implemented, right? If the button is always used to submit a form, that would be a test better suited for the form component